tests/nlu/data_augmentation/test_augmenter.py

```diff
@@ -0,0 +1,332 @@
+    )
+    report_file = os.path.join(
+        data_root, "data/test_nlu_paraphrasing/nlu_classification_report.json"
+    )
+    classification_report = rasa.shared.utils.io.read_json_file(report_file)
+    nlu_training_data = rasa.shared.nlu.training_data.loading.load_data(
+        os.path.join(data_root, "data/test_nlu_paraphrasing/nlu_train.yml")
+    )
+    intent_proportion = 0.5
+
+    intents_to_augment = _collect_intents_for_data_augmentation(
+        nlu_training_data=nlu_training_data,
+        intent_proportion=intent_proportion,
+        classification_report=classification_report,
+    )
+
+    num_intents = len(nlu_training_data.number_of_examples_per_intent)
+    num_intents_to_augment_per_criterion = round(num_intents * intent_proportion)
+    assert len(intents_to_augment) >= num_intents_to_augment_per_criterion
+    assert len(intents_to_augment) <= num_intents
+
+
+def test_augmenter_paraphrase_pool_creation(run: Callable[..., RunResult],):
+    from rasa.nlu.data_augmentation.augmenter import _create_paraphrase_pool
+
+    data_root = os.path.dirname(
+        os.path.dirname(os.path.dirname(os.path.dirname(__file__)))
+    )
+
+    paraphrases = rasa.shared.nlu.training_data.loading.load_data(
+        os.path.join(data_root, "data/test_nlu_paraphrasing/paraphrases.yml")
+    )
+
+    # See the file `rasa/data/test_nlu_paraphrasing/nlu_train.yml` for available intents
+    intents_to_augment = {"ask_transfer_charge", "check_earnings", "check_recipients"}
+
+    # Max sim score in `rasa/data/test_nlu_paraphrasing/paraphrases.yml` is 0.975
+    # Min sim score in `rasa/data/test_nlu_paraphrasing/paraphrases.yml` is 0.389
+
+    pool = _create_paraphrase_pool(
+        paraphrases=paraphrases,
+        intents_to_augment=intents_to_augment,
+        min_paraphrase_sim_score=0.98,
+        max_paraphrase_sim_score=0.99,
+    )
+    assert len(pool) == 0
+
+    pool = _create_paraphrase_pool(
+        paraphrases=paraphrases,
+        intents_to_augment=intents_to_augment,
+        min_paraphrase_sim_score=0.0,
+        max_paraphrase_sim_score=0.99,
+    )
+    assert len(pool["check_earnings"]) == 18
+    assert len(pool["check_recipients"]) == 18
+    assert len(pool["ask_transfer_charge"]) == 6
+
+    pool = _create_paraphrase_pool(
+        paraphrases=paraphrases,
+        intents_to_augment=intents_to_augment,
+        min_paraphrase_sim_score=0.0,
+        max_paraphrase_sim_score=0.45,
+    )
+
+    assert len(pool["check_earnings"]) == 3
+    assert len(pool["check_recipients"]) == 8
+    assert len(pool["ask_transfer_charge"]) == 2
+
+    pool = _create_paraphrase_pool(
+        paraphrases=paraphrases,
+        intents_to_augment=intents_to_augment,
+        min_paraphrase_sim_score=0.4,
+        max_paraphrase_sim_score=0.5,
+    )
+    assert len(pool["check_earnings"]) == 5
+    assert len(pool["check_recipients"]) == 8
+    assert len(pool["ask_transfer_charge"]) == 1
```

Comment:
Parameterize the test with min, max thresholds and expected number of examples in the pool to reduce code duplication.